gplugins/design_recipe/DesignRecipe.py

```diff
@@ -1,81 +1,151 @@
-from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.config import logger
 from gdsfactory.path import hashlib
+from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.typings import ComponentFactory
+from pydantic import BaseModel
 
 import gplugins.design_recipe as dr
 
 
+class Setup:
+    """
+    A dynamic class to store any setup information
+
+    This class allows designers to arbitrarily add setup information.
+    """
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def __eq__(self, other):
+        if not isinstance(other, Setup):
+            # Don't attempt to compare against unrelated types
+            return NotImplemented
+        return self.__dict__ == other.__dict__
+
+
 class DesignRecipe:
     """
     A DesignRecipe represents a flow of operations on GDSFactory components,
     with zero or more dependencies. Note that dependencies are assumed to be independent,
-    dependent dependencies should be nested. When `eval()`ed, A DesignRecipe `eval()`s 
+    dependent dependencies should be nested. When `eval()`ed, A DesignRecipe `eval()`s
     its dependencies if they they've become stale,
     and optionally executes some tool-specific functionality.
     For example,an FdtdDesignRecipe might simulate its `component` in
     Lumerial FDTD to extract its s-parameters.
-    """
 
-    # This `DesignRecipe`s dependencies. These are assumed to be independent
-    dependencies: dr.ConstituentRecipes
-
-    # the hash of the system last time eval() was executed
-    last_hash: int
 
-    # The component this DesignRecipe operates on. This is not necessarily
-    # the same `component` refered to in the `dependencies` recipes.
-    component: Optional[Component] = None
+    Attributes:
+        dependencies: This `DesignRecipe`s dependencies, assumed to be independent
+        last_hash: The hash of the system last time eval() was executed.
+                   This is used to determine whether the recipe's configuration
+                   has changed, meaning it must be re-eval'ed.
+        cell: GDSFactory layout cell or component. This is not necessarily the
+              same `component` referred to in the `dependencies` recipes.
+        layer_stack: PDK layerstack
+        dirpath: Root directory where recipes runs are stored
+        recipe_dirpath: Recipe directory where recipe results are stored.
+                        This is only created upon eval of the recipe.
+        run_convergence: Run convergence if True. Accurate simulations come from
+                         simulations that have run convergence.
+        override_recipe: Overrides recipe results if True. This runs the design
+                         recipe regardless if results are available and overwrites
+                         recipe results.
+        recipe_results: A dynamic object used to store recipe results and the
```

Comment:
what does 'dynamic object' mean? maybe an implementation-dependent object containing the recipe results?

Reply:
Dynamic meaning, users can arbitrarily add attributes to the Results instance. "Inheriting from Results" is appropriate.

Reply:
I think we had this conversation (prefer structured datatypes over dicts with arbitrary keys)?  correct me if I'm wrong